.github/workflows/stale-repos.yml

```diff
@@ -1,19 +1,47 @@
 on:
   schedule:
-  - cron: "0 */6 * * *" # every 6 hours
+    - cron: "0 */6 * * *" # every 6 hours
+  pull_request:
+    paths:
+      - 'configs/*.json'
+
+name: Check Stale Repos
 
 jobs:
   matrix:
     name: Find stale repositories
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
+      - id: repos
+        name: find repositories to check
+        run: |
+          # retrieves names of target repositories per JSON config file
+          # returns a JSON object with the following schema:
+          #  { "$JSON_CONFIG_FILE_PATH": ["$GITHUB_REPOSITORY_FULL_NAME"] }
+          repos="$(jq -nc '[inputs | {"\(input_filename)":(.repositories | map(.target))}] | add' configs/*.json)"
+
+          if [[ '${{ github.event_name }}' == 'pull_request' ]]; then
+            # resets the checkout to PR target commit
+            git fetch origin ${{ github.event.pull_request.base.sha }}
+            git reset --hard ${{ github.event.pull_request.base.sha }}
+
+            # retrieves names of target repositories per JSON config file
+            #  on PR target commit
+            base="$(jq -nc '[inputs | {"\(input_filename)":(.repositories | map(.target))}] | add' configs/*.json)"
+
+            # removes names of target repositories that were present in JSON config files
+            #  on PR target commit from the JSON object created on PR head commit
+            repos="$(jq -nc '$head | to_entries | map({"key":"\(.key)","value":(.value - ($base["\(.key)"] // []))}) | from_entries' --argjson head "$repos" --argjson base "$base")"
```

Comment:
That seems a bit convoluted. Isn't there an easier way to diff two jsons?

There are a number of command line tools to do so, and (since we strive to keep 3rd party dependencies to a minimum) there are also numerous ways to use `jq`, e.g. https://stackoverflow.com/questions/31930041/using-jq-or-alternative-command-line-tools-to-compare-json-files.

Reply:
It would have been even clearer if we didn't care about from which config file the repository comes from but I think it's useful info to have in the output.

I gave it another shot here: https://github.com/protocol/.github/pull/297/commits/c67a39d203102811be2a758f4c99adae27ec2bc8

Now this part reads as follows: https://github.com/protocol/.github/blob/c67a39d203102811be2a758f4c99adae27ec2bc8/.github/workflows/stale-repos.yml#L33